kernel/src/hil/date_time.rs

```diff
@@ -0,0 +1,77 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! HIL for Date Time interface
+
+use crate::ErrorCode;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DayOfWeek {
+    Sunday,
+    Monday,
+    Tuesday,
+    Wednesday,
+    Thursday,
+    Friday,
+    Saturday,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Month {
+    January,
+    February,
+    March,
+    April,
+    May,
+    June,
+    July,
+    August,
+    September,
+    October,
+    November,
+    December,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct Date {
+    pub year: u16,
+    pub month: Month,
+    pub day: u8,
+    pub day_of_week: DayOfWeek,
```

Comment:
This might be hard for the caller to provide when setting the datetime?

Reply:
But what if you want to use both date and time, you have to know the day of the week too. And what if an RTC provides date and time, but not day of week? I just wonder if making it an `Option<DayOfWeek>` would be more portable.

Reply:
From what I understood, the board RTC will update the DayOfWeek regardless of whether it was set or not. If the user doesn't need to use the DayOfWeek field he can simply ignore it, but in the back, the board will still update the DayOfWeek registers beginning with the value 0 (which is either Monday, Saturday or null depending on the board). I do not think making it Option<DayOfWeek> has any real advantage.

Reply:
Ok I just googled for "low power rtc" and the first three results from three different companies all support day of the week. Please ignore my original comment.